### DISCUSSION TITLE
你真的需要一个（专门的）向量数据库么 | 高策

### DISCUSSION THREAD

**Discussion by giscus** [gaocegege/blog-comments#17] [General]
# 你真的需要一个（专门的）向量数据库么 | 高策

江湖小虾米

http://gaocegege.com/Blog/vector

**Comment by Harry-Chen** [comment]
> * 将 prompt 向量进行搜索，寻找到最相似的 chunk 向量。
> * 将最相似的 chunk 向量与 prompt 向量拼接，作为 LLM 的输入。

我理解搜索出来的应该是 chunk 的原始文本，询问 LLM 用的也应该是原始的 prompt 和 chunk 文本，而不是 embedded 向量（否则 LLM 怎么理解呢？）。

FYI，原文也是这么说的：

> Use question embedding vector to search chunks from vector database
> Combine chunks and questions as a prompt, feed it to GPT

**Reply by gaocegege** [reply]
是的，这里是 typo

**Comment by xiaoFine** [comment]
我在人脸查重和chat-pdf等项目中也遇到了类似的境况，vector database采用的是国内的milvus2.X版本。
milvus2.X开始持varchar类型的字段和hybrid_search(vector+non-vector)，在某种程度上解决了meta filtering的问题;
剩下的一致性问题我是采用了比较激进的做法，就是把把业务数据也存到了milvus，这样crud都是milvus单体解决了；

相对的，当时么有用pgvector的原因是 公司的dba不会安装插件，以及部分环境的pg是用的公有云rds，不知道对插件的支持如何。

一些项目经历的分享：D

**Reply by gaocegege** [reply]
有道理，公有云的 PG 可能是一个问题。他们不一定开放插件能力，下面的 PG 都可能是魔改的
